modules/kibana/src/test/java/org/elasticsearch/kibana/KibanaThreadPoolTests.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.kibana;
+
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.indices.SystemIndexThreadPoolTests;
+import org.elasticsearch.plugins.Plugin;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class KibanaThreadPoolTests extends SystemIndexThreadPoolTests {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Set.of(KibanaPlugin.class);
+    }
+
+    public void testKibanaThreadPool() {
+        runWithBlockedThreadPools(() -> {
+            // index documents
+            String idToDelete = client().prepareIndex(".kibana").setSource(Map.of("foo", "delete me!")).get().getId();
+            String idToUpdate = client().prepareIndex(".kibana").setSource(Map.of("foo", "update me!")).get().getId();
+
+            // bulk index, delete, and update
+            Client bulkClient = client();
+            BulkResponse response = bulkClient.prepareBulk(".kibana")
+                .add(bulkClient.prepareIndex(".kibana").setSource(Map.of("foo", "search me!")))
+                .add(bulkClient.prepareDelete(".kibana", idToDelete))
+                .add(bulkClient.prepareUpdate().setId(idToUpdate).setDoc(Map.of("foo", "I'm updated!")))
+                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                .get();
+            assertFalse(response.hasFailures());
+
+            // match-all search
+            var searchResponse = client().prepareSearch(".kibana").setQuery(QueryBuilders.matchAllQuery()).get();
+            assertThat(searchResponse.getHits().getHits().length, equalTo(2));
```

Comment:
You'll need to release the `searchResponse` here (consider using `org.elasticsearch.test.hamcrest.ElasticsearchAssertions#assertHitCount(org.elasticsearch.action.search.SearchRequestBuilder, long)` to do so automatically).

Reply:
That's worked for the Kibana tests, but there's still a leak on the test for whether the thread pools are actually blocked.

What I'm doing right now is setting a short timeout on the request so that it fails while its tasks are in the thread pool queue, but I assume this isn't the right way to clean up resources. I'll look around in existing tests for another way of doing things, but let me know if there's a well-known way of doing this.

Reply:
Ah right yeah the timed `get` in that test doesn't actually cancel the operation when it times out, so it will ignore, and therefore leak, the eventual response. I think something like this should work:

```
        var e3 = expectThrows(
            ElasticsearchTimeoutException.class,
            () -> PlainActionFuture.get(
                fut -> client().prepareSearch(USER_INDEX).setQuery(QueryBuilders.matchAllQuery()).execute(fut.map(SearchResponse::decRef)),
                25,
                TimeUnit.MILLISECONDS
            )
        );
```

However maybe it's best to just go back to filling up the threadpool queues and then verifying that the request is rejected. Once the threads are all blocked (according to the phaser) you can just submit no-op tasks until you get a rejection on each executor.

Reply:
I like that second suggestion, of filling up the queues and checking rejections. I've gone with that.

Thanks for helping me straighten things out here. I'm not seeing leaks anymore.